lib/absinthe/type/extensions.ex

```diff
@@ -0,0 +1,111 @@
+defmodule Absinthe.Type.Extensions do
+  use Absinthe.Schema.Notation
+
+  @moduledoc """
+  This module contains additional data types.
+
+  To use: `import_types Absinthe.Type.Extensions`.
+  """
+
+  scalar :datetime, name: "DateTime" do
+    description """
+    The `DateTime` scalar type represents a date and time in the UTC
+    timezone. The DateTime appears in a JSON response as an ISO8601 formatted
+    string, including UTC timezone ("Z").
+    """
+
+    serialize &DateTime.to_iso8601/1
+    parse parse_with([Absinthe.Blueprint.Input.DateTime], &parse_datetime/1)
+  end
+
+  scalar :naive_datetime, name: "NaiveDateTime" do
+    description """
+    The `Naive DateTime` scalar type represents a naive date and time without
+    timezone. The DateTime appears in a JSON response as an ISO8601 formatted
+    string.
+    """
+
+    serialize &NaiveDateTime.to_iso8601/1
+    parse parse_with([Absinthe.Blueprint.Input.NaiveDateTime], &parse_naive_datetime/1)
+  end
+
+  scalar :date do
+    description """
+    The `Date` scalar type represents a date. The Date appears in a JSON
+    response as an ISO8601 formatted string.
+    """
+
+    serialize &Date.to_iso8601/1
+    parse parse_with([Absinthe.Blueprint.Input.Date], &parse_date/1)
+  end
+
+  scalar :time do
+    description """
+    The `Time` scalar type represents a time. The Time appears in a JSON
+    response as an ISO8601 formatted string.
+    """
+
+    serialize &Time.to_iso8601/1
+    parse parse_with([Absinthe.Blueprint.Input.Time], &parse_time/1)
+  end
+
+  @spec parse_datetime(any) :: {:ok, DateTime.t} | :error
+  defp parse_datetime(value) when is_binary(value) do
+    case DateTime.from_iso8601(value) do
+      {:ok, datetime, _offset} -> {:ok, datetime}
```

Comment:
I have something similar in my code, but I also consider a non-zero offset to be an input error:

```elixir
      case DateTime.from_iso8601(string) do
        {:ok, dt, 0}      -> {:ok, dt}
        {:ok, _, _}       -> {:error, :unsupported_offset}
        {:error, reason}  -> {:error, reason}
      end
```

That policy might be too strict for a general implementation, but I think it's slightly more honest about how offset information is treated by the server.